server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -696,6 +696,15 @@ int getNumBuffered() {
         int getNumReducePhases() { return numReducePhases; }
     }
 
+    private int resolveTrackTotalHits(SearchRequest request) {
+        if (request.scroll() != null) {
+            // no matter what the value of track_total_hits is
+            return SearchContext.TRACK_TOTAL_HITS_ACCURATE;
+        }
+        return request.source() == null ? SearchContext.DEFAULT_TRACK_TOTAL_HITS_UP_TO : request.source().trackTotalHitsUpTo() == null ?
```

Comment:
is the `request.source() == null?` check necessary on line 704? Seems like it's redundant given the check on line 700

p.s: I'm a curious engineer looking to learn more about elasticsearch's internals. I'm reading the diffs in my spare time to learn how experts in the field are building real-world distributed systems. Thanks for taking the time to answer my questions!

Reply:
The check is on `request.scroll` which is different than `request.source` so it is necessary.